src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java

```diff
@@ -0,0 +1,165 @@
+    private Map<ResourceLocation, ItemPropertyFunction> currentItemOverrides = Collections.emptyMap();
+
+    public ItemIconWithProperties()
+    {
+        super();
+    }
+
+    public ItemIconWithProperties(final PaneParams paneParams)
+    {
+        super(paneParams);
+
+        final String data = paneParams.getString(PARAM_PROPERTIES);
+        if (data != null && itemStack != null)
+        {
+            final CompoundTag tag;
+            try
+            {
+                tag = TagParser.parseTag(data);
+            }
+            catch (CommandSyntaxException e)
+            {
+                throw new RuntimeException(data, null);
+            }
+            tag.getAllKeys().forEach(itemKey -> {
+                if (tag.contains(itemKey, Tag.TAG_COMPOUND))
+                {
+                    final CompoundTag child = tag.getCompound(itemKey);
+                    final var itemOverrides = NBT_GENERIC_KEY.equals(itemKey) ? genericPropertyOverrides :
+                        itemPropertyOverrides.computeIfAbsent(NBT_CURRENT_ITEM.equals(itemKey) ? itemStack.getItem() :
+                            ForgeRegistries.ITEMS.getValue(ResourceLocation.tryParse(itemKey)), i -> new HashMap<>());
+
+                    child.getAllKeys().forEach(key -> {
+                        if (child.contains(key, Tag.TAG_ANY_NUMERIC))
+                        {
+                            final float value = child.getFloat(key); // intentionally ouf of lambda
+                            itemOverrides.put(ResourceLocation.tryParse(key), (itemStack, level, entity, seee) -> value);
+                        }
+                    });
+                }
+            });
+
+            onItemUpdate();
+        }
+    }
+
+    protected static String getNextItemKey(final String str, final int startIndex)
+    {
+        return str.substring(startIndex, str.indexOf('(', startIndex)).trim();
+    }
+
+    protected static String getNextItemOverrides(final String str, final int startIndex)
+    {
+        return str.substring(startIndex, str.indexOf(')', startIndex));
+    }
+
+    /**
+     * Short call for adding itemProperty to current item
+     */
+    public void addPropertyForCurrentItem(final ResourceLocation propertyKey, final ClampedItemPropertyFunction property)
+    {
+        itemPropertyOverrides
+            .computeIfAbsent(Objects.requireNonNull(itemStack, "Call #setItem before this method").getItem(), item -> new HashMap<>())
+            .put(propertyKey, property);
+    }
+
+    /**
+     * @return modifiable all item-based overrides
+     */
+    public Map<Item, Map<ResourceLocation, ItemPropertyFunction>> getItemPropertyOverrides()
+    {
+        return itemPropertyOverrides;
+    }
+
+    /**
+     * @return modifiable generic overrides
+     */
+    public Map<ResourceLocation, ItemPropertyFunction> getGenericPropertyOverrides()
+    {
+        return genericPropertyOverrides;
```

Comment:
Unmodifiable map maybe?

Reply:
intentionally modifiable, serves also as setter, basically the class doesnt care what's in the map

Reply:
As we have an "add" function. Probably makes sense to be unmodifiable here

Reply:
The add is only for non-generic and is pure syntax sugar (doesn't do anything special), I feel like unmodifiable makes little sense - you get full map methods vs we create all delegate method for both maps